### DISCUSSION TITLE
Define the coordinates of the elements for a 2 element mesh in the input file

### DISCUSSION THREAD

**Discussion by ambehnam** [idaholab/moose#21671] [Q&A Meshing] [answered]
Hi there. I want to validate an interface kernel by creating a 2 element mesh with arbitrary coordinates. currently I am generating the mesh using the following input commands:
```
[Mesh]
  [./gen]
    type = GeneratedMeshGenerator
    dim = 3
    nx = 2
    xmax = 2
    ny = 1
    ymax = 1
    nz = 1
    zmax = 1
    elem_type = HEX8
  [../]
  [./left]
    type = SubdomainBoundingBoxGenerator
    input = 'gen'
    block_id = 1
    bottom_left = '0 0 0'
    top_right = '1 1 1'
  [../]
  [./right]
    type = SubdomainBoundingBoxGenerator
    input = 'left'
    block_id = 2
    bottom_left = '1 0 0'
    top_right = '2 1 1'
  [../]
  [./breakmesh]
    type = BreakMeshByBlockGenerator
    input = right
  []
[]
```
but, I want to change the coordinate of the middle node slightly to activate the off diagonal parameters of a penalty tensor. So is there anyway to define the coordinates of these two elements in the input file?

Thanks,
Amir

**Comment by WilkAndy** [comment]
Some random remarks....   Do you have to use MOOSE to create your mesh?  If yes, and you're doing some sort of parametric study where the node position varies between multiple MOOSE simulations, then i think i'd write my own mesh generator.  Otherwise, then external meshers allow this sort of freedom, so i think i'd use one of those.

**Comment by GiudGiud** [comment, marked as answer]
Hello

I see a few options within MOOSE mesh generation have a "slightly off" middle node.
- add MoveNodeGenerator after those 3 https://mooseframework.inl.gov/moose/source/meshgenerators/MoveNodeGenerator.html
OR
- use a CartesianMG instead of GeneratedMG then use different lengths for two cells
https://mooseframework.inl.gov/moose/source/meshgenerators/CartesianMeshGenerator.html

Guilllaume
